services/tcpoverrpc/client/client.go

```diff
@@ -0,0 +1,128 @@
+/* Copyright (c) 2024 Snowflake Inc. All rights reserved.
+
+   Licensed under the Apache License, Version 2.0 (the
+   "License"); you may not use this file except in compliance
+   with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing,
+   software distributed under the License is distributed on an
+   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+   KIND, either express or implied.  See the License for the
+   specific language governing permissions and limitations
+   under the License.
+*/
+
+// Package client provides the client interface for 'tcpoverrpc'
+package client
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"net"
+	"os"
+	"strconv"
+
+	"github.com/google/subcommands"
+
+	"github.com/Snowflake-Labs/sansshell/client"
+	pb "github.com/Snowflake-Labs/sansshell/services/tcpoverrpc"
+	"github.com/Snowflake-Labs/sansshell/services/util"
+)
+
+const subPackage = "tcpoverrpc"
+
+func init() {
+	subcommands.Register(&tcpCmd{}, subPackage)
```

Comment:
Just some thoughts on how to structure this functionality in the `sanssh` cmd/subcmd hierarchy....

I know we have a httpoverrpc service, but I feel your tcpoverrpc differs from that model;  httpoverrpc is still interacting with the target host itself.  While your tcpoverrpc is trying to interact with other hosts from sanshell target host.  Based on that, I wonder if we can structure this differently, for example:

1.   as a subcommand of the existing "healthcheck" cmd?
2.  or, start a new top cmd "outbound" or "networking", subcmd: "tcping", to make it more clear the action;  also "outbound" implies more ways to extend with subcommands, say "udpping", or many other protocol layer health checks, icmp/ping, arp etc...